test/functional/tests/io/trim/test_trim.py

```diff
@@ -0,0 +1,252 @@
+       requests were sent at all and only to the first partition.
+    pass_criteria:
+      - Partition used for cache is discarded.
+      - Second partition is untouched - written pattern is preserved.
+    """
+    with TestRun.step("Clearing dmesg"):
+        TestRun.executor.run_expect_success("dmesg -C")
+
+    with TestRun.step("Preparing cache device"):
+        dev = TestRun.disks['cache']
+        dev.create_partitions([Size(500, Unit.MebiByte), Size(500, Unit.MebiByte)])
+        cas_part = dev.partitions[0]
+        non_cas_part = dev.partitions[1]
+
+    with TestRun.step("Writing different pattern on partitions"):
+        cas_fio = write_pattern(cas_part.system_path)
+        non_cas_fio = write_pattern(non_cas_part.system_path)
+        cas_fio.run()
+        non_cas_fio.run()
+
+    # TODO add blktracing for non-cas part
+    with TestRun.step("Starting blktrace against first (cache) partition"):
+        blktrace = BlkTrace(cas_part, BlkTraceMask.discard)
+        blktrace.start_monitoring()
+
+    with TestRun.step("Starting cache"):
+        cache = casadm.start_cache(cas_part, force=True)
+        metadata_size = get_metadata_size_from_dmesg()
+
+    with TestRun.step("Stop blktrace and check if discard requests were issued"):
+        cache_reqs = blktrace.stop_monitoring()
+        cache_part_start = cas_part.begin
+
+        # CAS should discard cache device during cache start
+        if len(cache_reqs) == 0:
+            TestRun.fail("No discard requests issued to the cas partition!")
+
+        non_meta_sector = (cache_part_start + metadata_size).get_value(Unit.Blocks512)
+        non_meta_size = (cas_part.size - metadata_size).get_value(Unit.Byte)
+        for req in cache_reqs:
+            if req.sector_number != non_meta_sector:
+                TestRun.fail(f"Discard request issued to wrong sector: {req.sector_number}, "
+                             f"expected: {non_meta_sector}")
+            if req.byte_count != non_meta_size:
+                TestRun.fail(f"Discard request issued with wrong bytes count: {req.byte_count}, "
+                             f"expected: {non_meta_size} bytes")
+
+        cas_fio.read_write(ReadWrite.read)
+        non_cas_fio.read_write(ReadWrite.read)
+        cas_fio.verification_with_pattern("0x00")
+        cas_fio.offset(metadata_size)
+        cas_fio.run()
+        non_cas_fio.run()
+
+    with TestRun.step("Stopping cache"):
+        cache.stop()
+
+
+@pytest.mark.parametrizex("cache_mode", CacheMode.with_traits(CacheModeTrait.InsertWrite))
+@pytest.mark.parametrizex("filesystem", Filesystem)
+@pytest.mark.parametrizex("cleaning_policy", CleaningPolicy)
+@pytest.mark.parametrizex("discard_support", [{"cache": False, "core": True},
+                                              {"cache": True, "core": False},
+                                              {"cache": True, "core": True}])
```

Comment:
At this moment this cannot be dict (see: https://github.com/Open-CAS/test-framework/issues/210). For now You should change it to some immutable structure, e.g. tuple. And don't forget to update usage of `discard_support` according to changes.